sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala

```diff
@@ -118,6 +118,18 @@ private[sql] object ArrowUtils {
             nullable = false,
             timeZoneId,
             largeVarTypes)).asJava)
+      case CalendarIntervalType =>
+        val fields = Seq(
```

Comment:
Hm, can we try to find the equivalent type in Arrow? e.g., https://arrow.apache.org/docs/cpp/api/datatype.html#_CPPv4N5arrow4Type4type23INTERVAL_MONTH_DAY_NANOE

Reply:
yes, `org.apache.arrow.vector.IntervalMonthDayNanoVector` and `org.apache.arrow.vector.PeriodDuration`

Reply:
But I found the `getInterval` in `ColumnVector.scala`, not sure whether we should follow it?

Reply:
I think no need to follow how we're doing with Spark internal format .. - we can just provide the input and output consistently .. how to handle shouldn't matter

Reply:
I mean, external ser/de with Arrow should use that corresponding types ideally. Internal ser/de we can use our existing `getInterval` approach from Arrow batch.